src/saturation/Saturation.c

```diff
@@ -0,0 +1,140 @@
+#include "Saturation.h"
+#include <math.h>
+
+double max(double x, double y);
+double min(double x, double y);
+double Hue(double red, double green, double blue, double cmax, double delta);
+double absolu(double x);
+void Saturation(Image* Bitmap, int x, int y,double sat);
+int Cap(double x);
+double CapSat(double x);
+
+void SaturationAbsolue(Image* Bitmap, double Indice)
+{
+  for(double x = 0; x < Bitmap->height; ++x)
+    {
+      for(double y = 0; y < Bitmap->width; ++y)
+	{
+	  Saturation(Bitmap, y, x, Indice);
+	}
+    }
+
+}
+
+
+void Saturation(Image* Bitmap, int x, int y,double sat)
+{
+  
+    double nrouge = Bitmap->pixels[x][y].red / 255.0;
+    double nvert = Bitmap->pixels[x][y].green / 255.0;
+    double nbleu = Bitmap->pixels[x][y].blue / 255.0;
+
+    double cmax = max(nbleu, max(nrouge,nvert));
+    double cmin = min(nbleu, min(nvert,nrouge));
+    double delta = cmax - cmin;
+
+    double hue = delta == 0 ? 0 : Hue(nrouge,nvert,nbleu,cmax,delta) ;
+    double luminance = (cmax + cmin)/2.0;
+    double saturation = delta == 0 ? 0 : CapSat(delta /(1-absolu(2*luminance-1))) ;
+
+    if(sat > 0)
+      saturation = CapSat(saturation + sat * (255 - saturation) * (saturation / 255.0));
+    else
+      saturation = CapSat(saturation + sat * saturation);
+  
+
+    double C = (1-absolu(2*luminance-1))*saturation;
+    double X = C * (1-absolu(fmod((hue/60),2) - 1));
+    double M = luminance - C/2;
+
+    if (hue >= 0 && hue < 60)
```

Comment:
Use if else instead of chained if to make it faster.
This function is called for every pixel of the image so it can have a significant impact.